scripts/prune_lm_dir.py

```diff
@@ -328,6 +351,153 @@ def FinalizeOutput(final_work_out):
         os.remove(args.lm_dir_out + "/num_splits")
 
 
+def MatchTargetSize(num_ngrams):
+    assert(args.target_num_ngrams > 0)
+    return abs(num_ngrams - args.target_num_ngrams) / float(args.target_num_ngrams) < args.tolerance
+
+def IterateOnce(scale, step, n_iter, recovery_step, recovery_size):
+    """Prune with a specific threshold and check whether the resulting LM matches the target-num-ngrams
+
+    Args:
+        scale: the threshold equals to scale * args.threshold
```

Comment:
This line 'scale: the threshold equals to scale * args.threshold' is not clear.  But it would be better if you just reorganize the code so that you can provide the actual threshold for pruning, and not have to explain this at all.